pkg/jobs/registry.go

```diff
@@ -476,6 +476,12 @@ func (r *Registry) isOrphaned(ctx context.Context, payload *jobspb.Payload) (boo
 			pendingMutations = hasAnyMutations || hasDropJob
 			return nil
 		}); err != nil {
+			if err == sqlbase.ErrDescriptorNotFound {
```

Comment:
errors.Is(err, sqlbase.ErrDescriptorNotFound) may be more resilient in case the error gets wrapped in the future.